TestLogic/crdb_internal/max_retry_counter test is flaky

I get this error after only a few seconds with `make stress PKG=./pkg/sql/logictest/... TESTS='TestLogic//crdb_internal'`:

```
    --- FAIL: TestLogic/local-opt (0.01s)
        --- FAIL: TestLogic/local-opt/crdb_internal (0.26s)
            --- FAIL: TestLogic/local-opt/crdb_internal/max_retry_counter (0.02s)
                logic.go:2336: 
                     
                    testdata/logic_test/crdb_internal:434: SELECT key, (max_retries > 1), flags LIKE '!%' AS f
                      FROM crdb_internal.node_statement_statistics
                     WHERE application_name = 'test_max_retry'
                    ORDER BY key, f
                    expected:
                        SELECT crdb_internal.force_retry(_)  true  false
                        SELECT crdb_internal.force_retry(_)  true  true
                        SET application_name = DEFAULT       false false
                    but found (query options: "") :
                        SELECT crdb_internal.force_retry(_)  false  false
                        SELECT crdb_internal.force_retry(_)  false  true
                        SET application_name = DEFAULT       false  false
```

--- a/pkg/sql/logictest/testdata/logic_test/crdb_internal
+++ b/pkg/sql/logictest/testdata/logic_test/crdb_internal
@@ -421,7 +421,7 @@ statement ok
 SET application_name = 'test_max_retry'
 
 statement OK
-SELECT crdb_internal.force_retry('2ms'::INTERVAL)
+SELECT crdb_internal.force_retry('50ms'::INTERVAL)
 
 statement OK
 RESET application_name
@@ -429,9 +429,11 @@ RESET application_name
 # Note: two rows of output are expected: one for the statement that
 # failed with a retry error, and one for the final retry attempt that
 # succeeded without an error.
-# Wel also expect the RESET statement to not have any retry.
+# We also expect the RESET statement to not have any retry.
+# This tests that the retry counter is properly reset to 0
+# between statements.
 query TBB
-SELECT key, (max_retries > 1), flags LIKE '!%' AS f
+SELECT key, (max_retries > 0), flags LIKE '!%' AS f
   FROM crdb_internal.node_statement_statistics
  WHERE application_name = 'test_max_retry'
 ORDER BY key, f

